referenceframe/frame.go

```diff
@@ -445,3 +445,133 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
+		{Min: -orientationVector.Theta, Max: orientationVector.Theta}, // Theta
+	}
+	return &poseFrame{
+		&baseFrame{name, limits},
+		geometry,
+	}, nil
+}
+
+// Transform on the poseFrame acts as the identity function. Whatever inputs are given are directly translated
+// in a 7DoF pose. We note that theta should be in radians.
+func (pf *poseFrame) Transform(inputs []Input) (spatial.Pose, error) {
+	if err := pf.baseFrame.validInputs(inputs); err != nil {
+		return nil, NewIncorrectInputLengthError(len(inputs), 7)
+	}
+	return inputsToPose(inputs), nil
+}
+
+// Interpolate interpolates the given amount between the two sets of inputs.
+func (pf *poseFrame) Interpolate(from, to []Input, by float64) ([]Input, error) {
+	if err := pf.baseFrame.validInputs(from); err != nil {
+		return nil, NewIncorrectInputLengthError(len(from), len(pf.DoF()))
+	}
+	if err := pf.baseFrame.validInputs(to); err != nil {
+		return nil, NewIncorrectInputLengthError(len(to), len(pf.DoF()))
+	}
+	fromPose, err := pf.Transform(from)
+	if err != nil {
+		return nil, err
+	}
+	toPose, err := pf.Transform(to)
+	if err != nil {
+		return nil, err
+	}
+	interpolatedPose := spatial.Interpolate(fromPose, toPose, by)
+	return PoseToInputsRadians(interpolatedPose), nil
+}
+
+// Geometries returns an object representing the 3D space associeted with the staticFrame.
+func (pf *poseFrame) Geometries(inputs []Input) (*GeometriesInFrame, error) {
+	transformByPose, err := pf.Transform(inputs)
+	if err != nil {
+		return nil, err
+	}
+	if len(pf.geometries) == 0 {
+		return NewGeometriesInFrame(pf.name, []spatial.Geometry{}), nil
+	}
+	transformedGeometries := []spatial.Geometry{}
+	for _, geom := range pf.geometries {
+		transformedGeometries = append(transformedGeometries, geom.Transform(transformByPose))
+	}
+	return NewGeometriesInFrame(pf.name, transformedGeometries), nil
+}
+
+// DoF returns the number of degrees of freedom within a model.
+func (pf *poseFrame) DoF() []Limit {
+	return pf.limits
+}
+
+// MarshalJSON serializes a Model.
+func (pf *poseFrame) MarshalJSON() ([]byte, error) {
+	return nil, errors.New("serializing a poseFrame is currently not supported")
+}
+
+// InputFromProtobuf converts pb.JointPosition to inputs.
+func (pf *poseFrame) InputFromProtobuf(jp *pb.JointPositions) []Input {
+	n := make([]Input, len(jp.Values))
+	for idx, d := range jp.Values {
+		n[idx] = Input{utils.DegToRad(d)}
+	}
+	return n
+}
+
+// ProtobufFromInput converts inputs to pb.JointPosition.
+func (pf *poseFrame) ProtobufFromInput(input []Input) *pb.JointPositions {
+	n := make([]float64, len(input))
+	for idx, a := range input {
+		n[idx] = utils.RadToDeg(a.Value)
+	}
+	return &pb.JointPositions{Values: n}
+}
+
+// PoseToInputsRadians is a convience method for turning poses into inputs
+// We note that the orientation of the pose will be understood
+// as OrientationVectorRadians.
+func PoseToInputsRadians(p spatial.Pose) []Input {
```

Comment:
I could prefer to keep the name as is otherwise how do I know how what my pose's orientation should be in?
I could end up passing in pose as Euler angles and not know until i get an error.

Reply:
Theres only one value in the set of 7 that is even an angular measurement, I think this adds confusion that can be cleared up with a good comment on the function

Reply:
ok, it will be changed.